File: WebAsistencia/WebRH/MAU/ServicioDeSeguridad.js
var ServicioDeSeguridad = function (un_proveedor_ajax) {
    this.proveedor_ajax = un_proveedor_ajax;
};

ServicioDeSeguridad.prototype.getUsuarioPorIdPersona = function (id_persona, onSuccess, onError) {
    this.proveedor_ajax.postearAUrl({ url: "../AjaxWS.asmx/GetUsuarioPorIdPersona",
        data: {
            id_persona: id_persona
        },
        success: function (usuario_json) {
            var usuario = new Usuario({
                id: usuario_json.Id,
                nombre: usuario_json.Nombre,
                apellido: usuario_json.Apellido,
                legajo: usuario_json.Legajo,
                documento: usuario_json.Documento,
                alias: usuario_json.NombreDeUsuario
            });
            onSuccess(usuario);
        },
        error: function (XMLHttpRequest, textStatus, errorThrown) {
            onerror(errorThrown);
        }
    });
};

ServicioDeSeguridad.prototype.getPermisosPara = function (usuario, onSuccess, onError) {
    this.proveedor_ajax.postearAUrl({ url: "../AjaxWS.asmx/GetPermisosPara",
        data: {
            usuario: usuario.nombre
        },
        success: function (permisos) {
            onSuccess(permisos);
        },
        error: function (XMLHttpRequest, textStatus, errorThrown) {
            onerror(errorThrown);
        }
    });
};

ServicioDeSeguridad.prototype.concederPermisoA = function (usuario, funcionalidad, onSuccess, onError) {
    this.proveedor_ajax.postearAUrl({ url: "../AjaxWS.asmx/ConcederPermisoA",
        data: {
            usuario: usuario.nombre,
            funcionalidad: funcionalidad
        },
        success: function () {
            onSuccess();
        },
        error: function (XMLHttpRequest, textStatus, errorThrown) {
            onerror(errorThrown);
        }
    });
};

ServicioDeSeguridad.prototype.denegarPermisoA = function (usuario, funcionalidad, onSuccess, onError) {
    this.proveedor_ajax.postearAUrl({ url: "../AjaxWS.asmx/DenegarPermisoA",
        data: {
            usuario: usuario.nombre,
            funcionalidad: funcionalidad
        },
        success: function () {
            onSuccess();
        },
        error: function (XMLHttpRequest, textStatus, errorThrown) {
            onerror(errorThrown);
        }
    });

};
ServicioDeSeguridad.prototype.getFuncionalidades = function (onSuccess, onError) {
    this.proveedor_ajax.postearAUrl({ url: "../AjaxWS.asmx/GetFuncionalidades",
        data: {
        },
        success: function (funcionalidades) {
            onSuccess(funcionalidades);
        },
        error: function (XMLHttpRequest, textStatus, errorThrown) {
            onerror(errorThrown);
        }
    });
};

File: ProyectoNet/General/MAU/ServicioDeSeguridad.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using General;
using General.Repositorios;

namespace AdministracionDeUsuarios
{
    public class ServicioDeSeguridad
    {
        private List<Usuario> usuarios;
        private General.Repositorios.RepositorioDePersonas repositorioDePersonas;

        public ServicioDeSeguridad(RepositorioDePersonas repositorioDePersonas)
        {
            this.repositorioDePersonas = repositorioDePersonas;
        }

        public Usuario GetUsuarioPorIdPersona(int id_persona)
        {
            var persona = repositorioDePersonas.GetPersonas().Find(p=> p.Id==id_persona);
            var usuario = new Usuario()
            {
                Nombre = persona.Nombre,
                Apellido = persona.Apellido,
                Documento = persona.Documento,
                Legajo = persona.Legajo,
                Id = persona.Id,
                NombreDeUsuario = "prueba"
            };
            return usuario;
        }
    }
}
